include/openthread/dnssd_server.h

```diff
@@ -159,8 +183,10 @@ typedef struct otDnssdCounters
     uint32_t mNameErrorResponse;      ///< The number of name error responses
     uint32_t mNotImplementedResponse; ///< The number of 'not implemented' responses
     uint32_t mOtherResponse;          ///< The number of other responses
+    uint32_t mResolvedBySrp;          ///< The number of queries completely resolved by the local SRP server
 
-    uint32_t mResolvedBySrp; ///< The number of queries completely resolved by the local SRP server
+    otUpstreamDnsCounters mHandledByUpstream; ///< The number of successful / failed queries that
```

Comment:
Since `mHandledByUpstream` is introduced as a member of `otDnssdCounters`, I'd suggest to explicitly document the relationship of the counters. E.g. whether a successful upstream DNS query will also count to  `mSuccessResponse`.